docs/decisions/007-cluster-mtls.md

```diff
@@ -0,0 +1,188 @@
+* **ExpandSecret**: Executors request secret values from the scheduler's secret store. This is the most sensitive RPC—secrets should only be available to authenticated cluster members.
+* **SchedulerGrpcServer**: Ballista's task dispatch protocol for scheduling query fragments to executors.
+
+mTLS provides mutual authentication (both parties verify identity) and transport encryption, ensuring:
+
+1. Only authenticated nodes can join the cluster and receive secrets.
+2. All traffic is encrypted, preventing interception of secrets or query data.
+3. No rogue processes can impersonate executors to exfiltrate data.
+
+Related decisions:
+
+* [DR-004: Use Apache Ballista as Spice's distributed query framework](./004-distributed-query-framework.md)
+* [DR-005: Extend Apache Ballista with Remote Catalog, UDF Sync, and Cluster Security](./005-ballista-extensions.md)
+* [DR-006: High Availability Distributed Query with Stateless Schedulers](./006-ha-distributed-query.md)
+
+## Design Principles
+
+* **Secure by default**: mTLS is required for cluster mode unless explicitly opting out with `--allow-insecure-connections`.
+* **Developer experience first**: CLI tooling (`spice cluster tls init`, `spice cluster tls add`) simplifies certificate generation; `--allow-insecure-connections` enables quick dev/test setup without certificates.
+* **Simplicity**: Standard TLS/mTLS patterns (three-file configuration) work with existing certificate management tools.
+* **Industry standards**: gRPC with `tonic` TLS support; standard X.509 certificates; SANs for hostname verification.
+* **Extensibility**: Certificate paths are CLI arguments, allowing integration with any certificate management solution.
+
+## Design
+
+### Port Separation
+
+| Port  | Visibility | Services                                            | mTLS Required              |
+| ----- | ---------- | --------------------------------------------------- | -------------------------- |
+| 50051 | Public     | `FlightServiceServer` (user queries), `OtelService` | Optional (user TLS config) |
+| 8090  | Public     | HTTP API (REST queries, health, status)             | Optional (user TLS config) |
+| 9090  | Public     | Prometheus metrics                                  | No                         |
+| 50052 | Internal   | `SchedulerGrpcServer`, `ClusterService`             | **Required**               |
+
+### CLI Arguments
+
+```bash
+# Role selection
+--role {scheduler|executor}               # Explicit role (executor implied if --scheduler-address set)
+
+# mTLS configuration (required unless --allow-insecure-connections)
+--node-mtls-ca-certificate-file <path>    # CA cert to validate peer certificates
+--node-mtls-certificate-file <path>       # This node's certificate
+--node-mtls-key-file <path>               # This node's private key
+
+# Network configuration
+--node-advertise-address <host:port>      # How other nodes reach this node
```

Comment:
```suggestion
--node-advertise-address <host>      # How other nodes reach this node
```

Part of my simplification effort is to remove the port here - as a reference, the Kubernetes API server doesn't allow configuring a separate port for the advertise address and just uses it from the bind address.